### PR TITLE
Some initial bits for tracer cell face vertical integrals

### PR DESCRIPTION
**Pull request by christophernhill**
@glwagner and @simone-silvestri - should we merge this with global-lat-lon? 

It is based on that and the most recent immersed seems to be in global-lat-lon.

### CODE REVIEW

**Comment by glwagner** [comment]
I think we want to introduce this major change to operators in a separate PR right? We also have to change derivatives, abstract operations, and grid metric operations?

**Comment by christophernhill** [comment]
> I think we want to introduce this major change to operators in a separate PR right? We also have to change derivatives, abstract operations, and grid metric operations?

@glwagner separate PR is fine too. I haven't been keeping up with where definitive immersed_boundary bits are. It seemed like there were some important updates in global-lat-lon. I am not clear if it can be a PR against main, since main may be missing some immersed bc bits this uses.

There are still some bits to work on after the bits in this PR, to get everything fully consistent. I am surprised things run as well as they do at the moment!! I think somehow we must end up with the bathymetry shifted a bit weird. It would be nice to fix, both implicit solve and split explicit are going to struggle I think without this patched up somehow.

**Comment by glwagner** [comment]
One can use your quick patch to see if it fixes the implicit solve, no problem.

The bulk of the work needed to solve this issue is independent of immersed boundaries (eliminating usage of dz^aac etc in favor of new metrics everywhere in the code). This requires widespread changes.

**Comment by glwagner** [comment]
Sorry, I was confused when I commented on the above PR (blame the github iOS app...), I thought I was commenting on issue #2049 .

File: src/ImmersedBoundaries/grid_fitted_immersed_boundaries.jl [hunk 1]
```diff
@@ -57,6 +57,18 @@ const GFBIBG = ImmersedBoundaryGrid{<:Any, <:Any, <:Any, <:Any, <:Any, <:GridFit
                                              zero(eltype(ibg.grid)),
                                              Δzᵃᵃᶜ(i, j, k, ibg.grid))
 
+@inline Δzᶠᶜᶜ(i, j, k, ibg::GFBIBG) = ifelse(  is_immersed(i-1, j  , k, ibg.grid, ibg.immersed_boundary)
+                                             | is_immersed(i  , j  , k, ibg.grid, ibg.immersed_boundary)
+                                             ,
+                                             zero(eltype(ibg.grid)),
+                                             Δzᵃᵃᶜ(i, j, k, ibg.grid))
```
Comment (glwagner) [inline src/ImmersedBoundaries/grid_fitted_immersed_boundaries.jl:64]:
```suggestion
@inline Δzᶠᶜᶜ(i, j, k, ibg::GFBIBG) = ifelse(solid_node(Face(), Center(), Center(), i, j, k, ibg), 
                                             zero(eltype(ibg)),
                                             Δzᵃᵃᶜ(i, j, k, ibg.grid))
```

File: src/ImmersedBoundaries/grid_fitted_immersed_boundaries.jl [hunk 2]
```diff
@@ -57,6 +57,18 @@ const GFBIBG = ImmersedBoundaryGrid{<:Any, <:Any, <:Any, <:Any, <:Any, <:GridFit
...
+
+@inline Δzᶜᶠᶜ(i, j, k, ibg::GFBIBG) = ifelse(  is_immersed(i  , j-1, k, ibg.grid, ibg.immersed_boundary)
+                                             | is_immersed(i  , j  , k, ibg.grid, ibg.immersed_boundary)
+                                             ,
+                                             zero(eltype(ibg.grid)),
+                                             Δzᵃᵃᶜ(i, j, k, ibg.grid))
```
Comment (glwagner) [inline src/ImmersedBoundaries/grid_fitted_immersed_boundaries.jl:70]:
```suggestion
@inline Δzᶜᶠᶜ(i, j, k, ibg::GFBIBG) = ifelse(solid_node(Center(), Face(), Center(), i, j, k, ibg),
                                             zero(eltype(ibg)),
                                             Δzᵃᵃᶜ(i, j, k, ibg.grid))
```

File: src/Operators/spacings_and_areas_and_volumes.jl [hunk 3]
```diff
@@ -46,6 +46,9 @@ The operators in this file fall into three categories:
 @inline Δzᵃᵃᶜ(i, j, k, grid::RegularRectilinearGrid) = grid.Δz
 @inline Δzᵃᵃᶜ(i, j, k, grid::VerticallyStretchedRectilinearGrid) = @inbounds grid.Δzᵃᵃᶜ[k]
 
+@inline Δzᶠᶜᶜ(i, j, k, grid::RegularRectilinearGrid) = grid.Δz
+@inline Δzᶜᶠᶜ(i, j, k, grid::VerticallyStretchedRectilinearGrid) = @inbounds grid.Δzᵃᵃᶜ[k]
```
Comment (glwagner) [inline src/Operators/spacings_and_areas_and_volumes.jl:50]:
```suggestion
@inline Δzᶠᶜᶜ(i, j, k, grid) = Δzᵃᵃᶜ(i, j, k, grid)
@inline Δzᶜᶠᶜ(i, j, k, grid) = Δzᵃᵃᶜ(i, j, k, grid)
```

**Review by glwagner — APPROVED** [review]
After the suggested changes this looks like a good start.

**Pull request merged**